### PR TITLE
WT-8640 Upgrading the variant for cppsuite stress tests and tuning some configurations

### PR DESCRIPTION
**Pull request by luke-pearson**

### CODE REVIEW

File: test/evergreen.yml [hunk 1]
```diff
@@ -347,15 +348,57 @@ functions:
...
+      params:
+        shell: bash
+        script: |
+          set -o verbose
+          if [ -f wiredtiger/cmake_build/test/cppsuite/cppsuite_exit_code ]; then
+            exit_code=`cat wiredtiger/cmake_build/test/cppsuite/cppsuite_exit_code`
+          else
+            exit_code=0
+          fi
+          rm -rf wiredtiger
```
Comment (lukech) [inline test/evergreen.yml:400]: Is it worth adding a comment to describe the reason for the folder removal?

Comment (luke-pearson) [reply test/evergreen.yml:400]: Will do.

Comment (luke-pearson) [reply test/evergreen.yml:401]: 78893f624

File: test/evergreen.yml [hunk 2]
```diff
@@ -347,15 +348,57 @@ functions:
...
+        # their paths defined relative to this directory.
+        # The below script saves the exit code from the test to use it later in this function. By
+        # doing this we can define our own custom artifact upload task without it being cancelled by
+        # the test failing.
+        # Additionally if the test fails perf statistics won't be uploaded as they may be invalid
+        # due to the test failure.
+        working_dir: "wiredtiger/cmake_build/test/cppsuite"
+        shell: bash
+        script: |
+          set -o verbose
```
Comment (lukech) [inline test/evergreen.yml:362]: Guess it's intentional not to put `set -o errexit` here as you want perf and artifacts uploads to run through in the case of test failures.

**Review by lukech — APPROVED** [review]
I reviewed only the `evergreen.yml` changes. LGTM

File: test/evergreen.yml (hunk 2 above)
Comment (luke-pearson) [reply test/evergreen.yml:362]: Yeah set -o errexit breaks this script.

File: test/cppsuite/configs/burst_inserts_stress.txt [hunk 3]
```diff
@@ -1,31 +1,43 @@
-# Used as a stress test for the framework.
+# Burst inserts is attempting to replicate conditions seen in WT-7798.
+# This workload is a highly stressful insert and read workload that performs insertions rapidly for
+# burst duration and then sleeps the op_rate defined in the insert config.
```
Comment (etienneptl) [inline test/cppsuite/configs/burst_inserts_stress.txt:3]: `burst duration` -> Use `burst_duration` if you use `op_rate`.

File: test/cppsuite/configs/burst_inserts_stress.txt [hunk 4]
```diff
@@ -1,31 +1,43 @@
...
+        ops_per_transaction=(max=30,min=0),
+        thread_count=100,
+        value_size=1000000
+    ),
+    checkpoint_config=
+    (
+        op_rate=120s,
     )
 ),
+# Operation tracker is disabled as it would create a very large table that cannot be cleaned up.
```
Comment (etienneptl) [inline test/cppsuite/configs/burst_inserts_stress.txt:40]: Can we say it cannot be cleaned up because we only insert unique keys

File: test/cppsuite/configs/hs_cleanup_stress.txt [hunk 5]
```diff
@@ -11,40 +12,30 @@ statistics_config=
...
     (
-        max=170000,
-        min=150000,
-        postrun=false,
         save=true,
     ),
     stat_cache_size=
     (
-        max=110,
+        # one day maybe we'll stop blowing out the cache.
```
Comment (etienneptl) [inline test/cppsuite/configs/hs_cleanup_stress.txt:25]: Should we add a `FIXME-WT-9339`?

File: test/cppsuite/configs/hs_cleanup_stress.txt [hunk 6]
```diff
@@ -54,30 +45,30 @@ workload_manager=
...
+        ops_per_transaction=(max=10,min=0),
+        thread_count=100,
+        value_size=1000000
     )
 ),
 operation_tracker=
 (
-    enabled=true,
+    enabled=false,
     op_rate=20s
```
Comment (etienneptl) [inline test/cppsuite/configs/hs_cleanup_stress.txt:73]: We can remove this if the tracker is disabled

File: test/cppsuite/configs/operations_test_stress.txt [hunk 7]
```diff
@@ -1,14 +1,14 @@
...
 (
     stat_db_size=
     (
         #At the end of the run the data files are approximately 2.3MB each. Which is a total of:
         #1.15GB, the history store isn't significant. Give the workload an extra 200MB of margin.
-        max=1350000000,
+        # max=1350000000,
         # FIXME-WT-8886 - This check has been disabled to remove noisy failures in evergreen and
         # will be properly corrected in WT-8886.
         runtime=false,
```
Comment (etienneptl) [inline test/cppsuite/configs/operations_test_stress.txt:14]: Next line where `save` is set seems to have the wrong indentation

File: test/cppsuite/configs/operations_test_stress.txt [hunk 8]
```diff
@@ -18,58 +18,58 @@ metrics_monitor=
...
-        thread_count=25,
-        value_size=10000
+        ops_per_transaction=(max=10,min=0),
+        thread_count=40,
+        value_size=1000000
     )
 ),
 operation_tracker=
 (
     # FIXME-WT-8640 Enable the tracker again once we have moved to a different host or modified the
```
Comment (etienneptl) [inline test/cppsuite/configs/operations_test_stress.txt:70]: You should be able to remove the FIXME.
Did `s_mentions` tell you about this FIXME comment?

File: test/cppsuite/src/main/test.cpp [hunk 9]
```diff
@@ -94,9 +94,8 @@ test::run()
     bool enable_logging, statistics_logging;
     configuration *statistics_config;
     std::string statistics_type;
-    /* Build the database creation config string. */
-    std::string db_create_config = CONNECTION_CREATE;
-
+    /* Build the database creation config string. Allow for a maximum 1024 sessions. */
+    std::string db_create_config = CONNECTION_CREATE + ",session_max=1024";
```
Comment (etienneptl) [inline test/cppsuite/src/main/test.cpp:98]: Why do we need to set the max number of sessions?

**Review by etienneptl — CHANGES_REQUESTED** [review]
Looks good, I left a few comments!

File: test/cppsuite/configs/operations_test_stress.txt (hunk 8 above)
Comment (luke-pearson) [reply test/cppsuite/configs/operations_test_stress.txt:70]: Theres an issue `s_mentions` on my tree that makes it impossible to tell if it is throwing an error. I will create a jira to follow up.

File: test/cppsuite/src/main/test.cpp (hunk 9 above)
Comment (luke-pearson) [reply test/cppsuite/src/main/test.cpp:98]: By default its like 128 or something and if each thread has its own session we can easily go over that number. I bumped it to 1024 for now.

File: test/cppsuite/configs/operations_test_stress.txt (hunk 8 above)
Comment (luke-pearson) [reply test/cppsuite/configs/operations_test_stress.txt]: bde1106ec

File: test/cppsuite/configs/operations_test_stress.txt (hunk 7 above)
Comment (luke-pearson) [reply test/cppsuite/configs/operations_test_stress.txt:11]: Couldn't see what you were referring to here.

File: test/cppsuite/configs/hs_cleanup_stress.txt (hunk 6 above)
Comment (luke-pearson) [reply test/cppsuite/configs/hs_cleanup_stress.txt]: bde1106ec

File: test/cppsuite/configs/hs_cleanup_stress.txt (hunk 5 above)
Comment (luke-pearson) [reply test/cppsuite/configs/hs_cleanup_stress.txt:26]: bde1106ec

File: test/cppsuite/configs/burst_inserts_stress.txt (hunk 4 above)
Comment (luke-pearson) [reply test/cppsuite/configs/burst_inserts_stress.txt]: bde1106ec

File: test/cppsuite/configs/burst_inserts_stress.txt (hunk 3 above)
Comment (luke-pearson) [reply test/cppsuite/configs/burst_inserts_stress.txt]: bde1106ec

File: test/cppsuite/configs/operations_test_stress.txt (hunk 7 above)
Comment (etienneptl) [reply test/cppsuite/configs/operations_test_stress.txt:11]: Maybe it's GitHub that is showing a wrong indentation, that's fine.

File: test/cppsuite/configs/operations_test_stress.txt (hunk 8 above)
Comment (etienneptl) [reply test/cppsuite/configs/operations_test_stress.txt]: `s_mentions` cannot return an error, it simply prints stuff that you may want to look at.

File: dist/s_string.ok [hunk 10]
```diff
@@ -672,6 +672,7 @@ children's
 chk
 chmod
 chongo
+chrono
```
Comment (etienneptl) [inline dist/s_string.ok:675]: Why did you need to add this?

**Review by etienneptl — APPROVED** [review]
LGTM, one last comment regarding the changes in `s_string`. Also, make sure there is no more FIXME regarding WT-8640 but `s_mentions` should help you with that.

File: test/cppsuite/configs/operations_test_stress.txt (hunk 8 above)
Comment (luke-pearson) [reply test/cppsuite/configs/operations_test_stress.txt]: No I mean you can't tell if it genuinely found a problem, currently s_mentions dumps like 5 pages of information to my screen.

Comment (etienneptl) [reply test/cppsuite/configs/operations_test_stress.txt]: Sounds really weird, please add your output to the ticket and I can have a look.
